index.bs

```diff
@@ -1370,77 +880,818 @@ What configuration is the device in after it resets?
+
+        The {{USBDevice/configurations}} getter steps are:
+          1. Return {{USBDevice/[[configurations]]}}.
+
+    : <dfn>opened</dfn>
+    ::
+        The {{USBDevice/opened}} attribute SHALL be set to <code>true</code> when the
+        device is opened by the current execution context and SHALL be set to
+        <code>false</code> otherwise.
+</dl>
+
+### Methods ### {#usbdevice-methods}
+
+<dl dfn-type=method dfn-for="USBDevice">
+    : <dfn>open()</dfn>
+    ::
+        The {{USBDevice/open()}} method, when invoked, MUST return a new {{Promise}}
+        |promise| and run the following steps <a>in parallel</a>:
+
+          1. Let |device| be the target {{USBDevice}} object.
+          2. If |device| is no longer connected to the system, <a>reject</a> |promise|
+            with a {{NotFoundError}} and abort these steps.
+          3. If <code>|device|.{{USBDevice/opened}}</code> is <code>true</code>
+            <a>resolve</a> |promise| and abort these steps.
+          4. Perform the necessary platform-specific steps to begin a session with the
+            device. If these fail for any reason <a>reject</a> |promise| with a
+            {{NetworkError}} and abort these steps.
+          5. Set <code>|device|.{{USBDevice/opened}}</code> to <code>true</code> and
+            <a>resolve</a> |promise|.
+
+    : <dfn>close()</dfn>
+    ::
+        The {{USBDevice/close()}} method, when invoked, MUST return a new {{Promise}}
+        |promise| and run the following steps <a>in parallel</a>:
+
+          1. Let |device| be the target {{USBDevice}} object.
+          2. If |device| is no longer connected to the system, <a>reject</a> |promise|
+            with a {{NotFoundError}} and abort these steps.
+          3. If <code>|device|.{{USBDevice/opened}}</code> is <code>false</code>
+            <a>resolve</a> |promise| and abort these steps.
+          4. Abort all other algorithms currently running against this device and
+            <a>reject</a> their associated promises with an {{AbortError}}.
+          5. Perform the necessary platform-specific steps to release any claimed
+            interfaces as if {{USBDevice/releaseInterface(interfaceNumber)}} had been
+            called for each claimed interface.
+          6. Perform the necessary platform-specific steps to end the session with the
+            device.
+          7. Set <code>|device|.{{USBDevice/opened}}</code> to <code>false</code> and
+            <a>resolve</a> |promise|.
+
+        Note: When no [[!ECMAScript]] code can observe an instance of {{USBDevice}} |device|
+        anymore, the UA SHOULD run |device|.{{USBDevice/close()}}.
+
+    : <dfn>selectConfiguration(configurationValue)</dfn>
+    ::
+        <div algorithm="selectConfiguration">
+            When invoked, MUST return a new {{Promise}} |promise| and run the following steps in
+            parallel:
+            1. Let |configurationValue| be the parameter of <code>configurationValue</code>.
```

Comment:
This isn't step shouldn't be necessary. The variable *configurationValue* should link to the parameter from the method definition above if you write it as,

```
<dfn>selectConfiguration(|configurationValue|)</dfn>
```

Reply:
Actually, it kind of links to the issue of https://github.com/tabatkins/bikeshed/issues/2225. In order to get around the errors, I moved around as in https://github.com/chengweih001/webusb_sameobject_patch/commit/6b111b1b5a5f1955752f4084d552b023002f1d78.

And this move around ended up not wrapping the |configuration| parameter in dfn tag within algorithm tag, then bikeshed complains no reference for the variable as it ends up only be referenced once.

I will add ignore to those variable and remove those non-necessary parameter assignment lines, let me know if you think with this reason it is fine to keep those.

Reply:
Interesting. It feels like we're probably structuring this the wrong way and if we did it differently Bikeshed would help us here. @tabatkins, can you give us some advice?

Reply:
Yeah, as they said, there's a Bikeshed bug currently where wrapping markdown dt/dds in a div (that is a child of a dl) doesn't get detected and corrected properly, so it retains the generated dl wrapper and emits bad HTML. I'll be fixing it shortly; then the algorithm wrapper can be put back around the dt/dd pair.

Or you can structure the html to not use a dt/dd pairs for the operations, and instead just make them flat `<div algorithm>`s that name the operation being defined in an intro paragraph, like in <https://drafts.css-houdini.org/css-typed-om/#dom-cssnumericvalue-add>

Reply:
@reillyeon 

How do you think? The style of https://drafts.css-houdini.org/css-typed-om/#dom-cssnumericvalue-add might just go back to like we have in current web usb: https://wicg.github.io/webusb/#ref-for-dom-usbdevice-selectconfiguration%E2%91%A1.

Reply:
That style works for me.

Reply:
There's a nice bit of CSS at <https://github.com/w3c/css-houdini-drafts/blob/main/css-typed-om/Overview.bs#L24-L40> you can pull in to still make the algorithms easily distinguishable.

Reply:
Thank you for the recommendation, I refactor these dfn and algorithm tags in another commit https://github.com/WICG/webusb/pull/212/commits/0d0b99236064dc502080a4b5c351e3f5e095fe4f